From 7c8abbe89403c2fcb08e3fda42049a97187e53ab Mon Sep 17 00:00:00 2001
From: fujiwarat <takao.fujiwara1@gmail.com>
Date: Tue, 28 Feb 2023 17:46:10 +0900
Subject: [PATCH] src: Call IBUS_TYPE_EMOJI_DATA in ibus_init()

Seems the evaluation of IBUS_EMOJI_DATA() depends on the compiler
optimization.

BUG=https://github.com/ibus/ibus/issues/2476
---
 src/ibusshare.c | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/src/ibusshare.c b/src/ibusshare.c
index 340168c84..bb59f3cbf 100644
--- a/src/ibusshare.c
+++ b/src/ibusshare.c
@@ -308,12 +308,15 @@ ibus_init (void)
     IBUS_TYPE_TEXT;
     IBUS_TYPE_ATTRIBUTE;
     IBUS_TYPE_ATTR_LIST;
-    IBUS_TYPE_LOOKUP_TABLE;
     IBUS_TYPE_COMPONENT;
+    IBUS_TYPE_EMOJI_DATA;
     IBUS_TYPE_ENGINE_DESC;
+    IBUS_TYPE_LOOKUP_TABLE;
     IBUS_TYPE_OBSERVED_PATH;
     IBUS_TYPE_REGISTRY;
     IBUS_TYPE_X_EVENT;
+    IBUS_TYPE_UNICODE_BLOCK;
+    IBUS_TYPE_UNICODE_DATA;
     _ibus_register_resource ();
 }
 
